assayist/processor/container_go_analyzer.py

```diff
@@ -0,0 +1,185 @@
+        # Check we have access to the executables we need.
+        assert_command(self.BACKVENDOR)
+        assert_command(self.GOVERSION)
+
+        build_info = self.read_metadata_file(self.BUILD_FILE)
+        build_id = build_info['id']
+        if not self.is_container_build(build_info):
+            log.info(f'Skipping build {build_id} because the build is not a container')
+            return
+
+        # This container's build is assumed to exist since it is
+        # created by the main analyzer.
+        build = content.Build.nodes.get(id_=build_id)
+        source_locations = build.source_location.all()
+        try:
+            source_location = source_locations[0]
+        except IndexError:
+            log.error('Missing source location for container build')
+            return
+
+        # First process the source code that's directly available in
+        # the dist-git repository.
+        dist_git_excludes = [
+            # Files expected to be local and not upstream.
+            '.git*',
+            'Makefile',
+            'sources',
+            'Dockerfile*',
+            'additional-tags',
+            'content_sets.yml',
+            'container.yaml',
+            '.oit',
+            'public',
+            'scripts',
+        ]
+        srcdir = os.path.join(self.input_dir, self.SOURCE_DIR)
+        self._process_source_code(source_location, srcdir,
+                                  excludes=dist_git_excludes)
+
+        # Next process source code from archives (from 'rhpkg sources').
+        # Look for tar archives and zip archives.
+        tar_archives = glob(os.path.join(srcdir, '*.tar.*'))
+        zip_archives = glob(os.path.join(srcdir, '*.zip'))
+        archives = [(unpack_tar, archive) for archive in tar_archives]
+        archives += [(unpack_zip, archive) for archive in zip_archives]
+        for unpack, archive in archives:
+            with tempfile.TemporaryDirectory() as subsrc:
+                unpack(archive, subsrc)
+                self._process_source_code(source_location, subsrc)
+
+        # Now claim all the Go executables.
+        self._claim_go_executables()
+
+    def _process_source_code(self, source_location, srcdir, excludes=None):
+        """Run backvendor on the source code.
+
+        :param SourceLocation source_location: local source code DB node
+        :param str path: path to source code to examine
+        :param list excludes: list of globs to ignore
+        """
+        # Run backvendor.
+        with tempfile.NamedTemporaryFile(mode='wt') as excludes_file:
+            options = ['-debug', '-template', self.BACKVENDOR_TEMPLATE]
+            if excludes:
+                excludes_file.write(''.join('%s\n' % e for e in excludes))
+                excludes_file.flush()
+                options += ['-exclude-from', excludes_file.name]
+
+            cmd = [self.BACKVENDOR] + options + [srcdir]
+            log.info(f'Running {cmd}')
```

Comment:
Optional: It'd be nice if this was:
```python
log.info(f'Running "{" ".join(cmd)}"')
```

Reply:
Note that one of the parameters includes unquoted tab characters, so the output looks a little messy when converted to a string.